### PR TITLE
Introduce ExecutionPayloadParams concept to be able to better compose…

### PR DESCRIPTION
**Pull request by LukaszRozmej**
… and validate complex method signatures

SImplify the code a bit.

Tests are failing but it is problem with tests, need to be fixed in main branch as they might get refactored.

### CODE REVIEW

**Review by flcl42 — APPROVED** [review]

File: src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Paris.cs [hunk 1]
```diff
@@ -64,12 +65,24 @@ private async Task<ResultWrapper<ForkchoiceUpdatedV1Result>> ForkchoiceUpdated(F
...
-    private async Task<ResultWrapper<PayloadStatusV1>> NewPayload(ExecutionPayload executionPayload, int version)
+    private async Task<ResultWrapper<PayloadStatusV1>> NewPayload(IExecutionPayloadParams executionPayloadParams, int version)
     {
-        if (!executionPayload.ValidateParams(_specProvider, version, out string? error))
+        ExecutionPayload executionPayload = executionPayloadParams.ExecutionPayload;
+
+        if (!executionPayload.ValidateFork(_specProvider))
+        {
+            if (_logger.IsWarn) _logger.Warn($"The payload is not supported by the current fork");
+            return ResultWrapper<PayloadStatusV1>.Fail("unsupported fork", ErrorCodes.UnsupportedFork);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Paris.cs:75]: Open question, should we use `ErrorCodes.InvalidParams` if `version < 3`? Before version 3 there is no error code `UnsupportedFork`.

Comment (flcl42) [reply src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Paris.cs:75]: It's mentioned in https://github.com/ethereum/execution-apis/pull/426 for V2, an attempt to move towards more descriptive error codes.

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V3.cs [hunk 2]
```diff
@@ -43,7 +43,7 @@ public async Task NewPayloadV1_should_decline_post_cancun()
 
         ResultWrapper<PayloadStatusV1> errorCode = (await rpcModule.engine_newPayloadV1(executionPayload));
 
-        Assert.That(errorCode.ErrorCode, Is.EqualTo(ErrorCodes.InvalidParams));
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V3.cs:46]: I assume that we should return InvalidParams still here
So if you pass newPayloadV1 with post-cancun timestamp your validation should fail once checking that BlobVersionedHashes are null and it should give you InvalidParams
UnsupportedFork will be when you pass newPayloadV3 with pre-cancun timestamp

Comment (flcl42) [reply src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V3.cs:46]: The idea is to show this error if we have Cancun timestamp in V2 and pre-Cancun timestamp in V3

File: src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Paris.cs (hunk 1 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Paris.cs:75]: yes, I think so

**Pull request merged**